# Task 2: Part 1

# Thomas Fire Perimeter

Author: Rachel Swick

Github Link: https://github.com/rfswick/eds220-hwk4.git

## Purpose

To create a saved geometry file for the perimeter of the 2017 Thomas Fire in Santa Barbara County.

## Highlights

Major steps include understanding the properties of the California wildfire boundaries dataset, selecting for the Thomas Fire burn perimeter, and saving the perimeter as a shapefile in the repository's data file.

## Dataset Description

Historic California Fires GeoDatabase: This dataset includes information about all recorded wildifires within the state of California that, according to CalFire, burn "≥10 acres timber, ≥50 acres brush, ≥300 acres grass, damages or destroys three or more structures or does $300,000 worth of damage, or results in loss of life.". Accessed: November 21, 2024

## References

Historical Fire Perimeters. CalFire. Updated April 2023. https://www.fire.ca.gov/what-we-do/fire-resource-assessment-program/fire-perimeters 

## 1. Import Libraries

In [1]:
# Load libraries
import os 

import numpy as np
import geopandas as gpd

## 2. Import Data

In [2]:
# Load historic California fire boundary data
ca_fires = gpd.read_file(os.path.join('data', 'fire.gdb'))

ERROR 1: PROJ: proj_create_from_database: Open of /opt/anaconda3/share/proj failed


## 3. Data Exploration

In [3]:
# Display the first 5 rows of the dataframe
ca_fires.head()

YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  \
0  2023.0    CA    CDF     SKU  WHITWORTH  00004808   
1  2023.0    CA    LRA     BTU     KAISER  00010225   
2  2023.0    CA    CDF     AEU    JACKSON  00017640   
3  2023.0    CA    CDF     AEU     CARBON  00018821   
4  2023.0    CA    CDF     AEU    LIBERTY  00018876   

                                  IRWINID           ALARM_DATE  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}  2023-06-17T00:00:00   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}  2023-06-02T00:00:00   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}  2023-07-01T00:00:00   
3  {CB41DB0A-E4B1-489D-A4EA-738F2CD6DB3B}  2023-07-11T00:00:00   
4  {F83F70A4-07A7-40B8-BD51-10CCC1C30D63}  2023-07-11T00:00:00   

             CONT_DATE  C_METHOD  CAUSE  OBJECTIVE  GIS_ACRES COMPLEX_NAME  \
0  2023-06-17T00:00:00       1.0    5.0        1.0   5.729125         None   
1  2023-06-02T00:00:00       1.0    5.0        1.0  13.602385         None   
2  2023-07-02T00:00:00       1.0    2.0        1.0  27.814465         None   
3  2023-07-11T00:00:00       1.0    9.0        1.0  58.760227         None   
4  2023-07-12T00:00:00       1.0   14.0        1.0  70.978996         None   

  COMPLEX_ID COMMENTS FIRE_NUM  Shape_Length     Shape_Area  \
0       None     None     None    933.907127   23184.946763   
1       None     None     None    986.343353   55046.898792   
2       None     None     None   2115.572740  112561.147688   
3       None     None     None   1994.866526  237794.210367   
4       None     None     None   4008.305800  287241.810389   

                                            geometry  
0  MULTIPOLYGON (((-243242.555 394267.371, -24324...  
1  MULTIPOLYGON (((-167826.147 203275.612, -16782...  
2  MULTIPOLYGON (((-79176.443 31393.490, -78959.0...  
3  MULTIPOLYGON (((-85936.934 47921.886, -85856.4...  
4  MULTIPOLYGON (((-86427.836 26461.613, -86436.1...

In [4]:
# Determine the data types of the columns
ca_fires.dtypes

YEAR_            float64
STATE             object
AGENCY            object
UNIT_ID           object
FIRE_NAME         object
INC_NUM           object
IRWINID           object
ALARM_DATE        object
CONT_DATE         object
C_METHOD         float64
CAUSE            float64
OBJECTIVE        float64
GIS_ACRES        float64
COMPLEX_NAME      object
COMPLEX_ID        object
COMMENTS          object
FIRE_NUM          object
Shape_Length     float64
Shape_Area       float64
geometry        geometry
dtype: object

In [5]:
# Determine the size of the data frame
ca_fires.shape

(22261, 20)

In [6]:
# Determine the CRS 
print('CRS:', ca_fires.crs)

CRS: epsg:3310


In [7]:
# Determine whether the data is geographic or projected
print('Is the CRS projected?', ca_fires.crs.is_projected)
print('Is the CRS geographic?', ca_fires.crs.is_geographic)

Is the CRS projected? True
Is the CRS geographic? False


## 4. Data Exploration Summary

The `ca_fires` dataframe contains 22,261 rows and 20 columns. The column containing spatial information is labeled `geometry`. The CRS of the data is EPSG:3310. The CRS of the data is projected, not geographic.

## 5. Thomas Fire Boundary Data Selection

In [8]:
# Select Thomas Fire perimeter boundaries
thomas_fire = ca_fires[(ca_fires.YEAR_ == 2017.0) & (ca_fires.FIRE_NAME == 'THOMAS')]

## 6. Save Thomas Fire Boundary

In [9]:
# Save the Thomas Fire boundary as a shapefile
thomas_fire.to_file(os.path.join('data', 'thomas_fire.shp'))

/tmp/ipykernel_3716509/2109970276.py:2: UserWarning: Column names longer than 10 characters will be truncated when saved to ESRI Shapefile.
  thomas_fire.to_file(os.path.join('data', 'thomas_fire.shp'))


## 7. Reasoning for Selecting Shapefile Format

I chose a shapefile because I am familiar with working with them, and they easily carry over other attribute information along with the spatial data. In addition, they work across many data science platforms.